### PR TITLE
Added user CRUD operations.

### PR DESCRIPTION
**Pull request by adee26**

### CODE REVIEW

**Review by Grbogdan — APPROVED** [review]
👍

**Review by Grbogdan — APPROVED** [review]
👍

File: src/main/java/com/sda/auction/controllers/UserController.java [hunk 1]
```diff
@@ -1,13 +1,37 @@
...
 import lombok.AllArgsConstructor;
 import org.springframework.beans.factory.annotation.Autowired;
-import org.springframework.web.bind.annotation.RestController;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.List;
+import java.util.Optional;
 
 @RestController
+@RequestMapping("/api/v1/")
```
Comment (andrei-spinu) [inline src/main/java/com/sda/auction/controllers/UserController.java]: @RequestMapping("/api/v1/users")

File: src/main/java/com/sda/auction/controllers/UserController.java [hunk 2]
```diff
@@ -1,13 +1,37 @@
...
+    public Optional<User> findUserById(@PathVariable int id){
+        return userService.findById(id);
+    }
+
+    @GetMapping("/users")
+    public List<User> findAllUsers(){
+        return userService.findAllUsers();
+    }
+
+    @PostMapping("/users")
```
Comment (andrei-spinu) [inline src/main/java/com/sda/auction/controllers/UserController.java]: delete "/users"

File: src/main/java/com/sda/auction/controllers/UserController.java [hunk 3]
```diff
@@ -1,13 +1,37 @@
...
 @AllArgsConstructor(onConstructor = @__(@Autowired))
 public class UserController {
     private final UserService userService;
 
+    @GetMapping("/users/{id}")
+    public Optional<User> findUserById(@PathVariable int id){
+        return userService.findById(id);
+    }
+
+    @GetMapping("/users")
```
Comment (andrei-spinu) [inline src/main/java/com/sda/auction/controllers/UserController.java]: delete "/users"

File: src/main/java/com/sda/auction/controllers/UserController.java [hunk 4]
```diff
@@ -1,13 +1,37 @@
...
+import java.util.List;
+import java.util.Optional;
 
 @RestController
+@RequestMapping("/api/v1/")
 @AllArgsConstructor(onConstructor = @__(@Autowired))
 public class UserController {
     private final UserService userService;
 
+    @GetMapping("/users/{id}")
```
Comment (andrei-spinu) [inline src/main/java/com/sda/auction/controllers/UserController.java]: delete "/users"

File: src/main/java/com/sda/auction/controllers/UserController.java [hunk 5]
```diff
@@ -1,13 +1,37 @@
...
+    public List<User> findAllUsers(){
+        return userService.findAllUsers();
+    }
+
+    @PostMapping("/users")
+    public void saveUser(@RequestBody User user){
+        userService.createUser(user);
+    }
+
+    @PutMapping("/users/{id}")
```
Comment (andrei-spinu) [inline src/main/java/com/sda/auction/controllers/UserController.java]: delete "/users"

**Review by andrei-spinu — CHANGES_REQUESTED** [review]

**Review by andrei-spinu — APPROVED** [review]

**Pull request merged**